lib/experimental/html/class-wp-html-text-replacement.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * HTML Walker: Text replacement class.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ * @since 6.1.0
+ */
+
+/**
+ * Data structure used to replace existing content from start to end that allows to drastically improve performance.
+ *
+ * @since 6.1.0
+ *
+ * @see WP_HTML_Walker
+ */
+class WP_HTML_Text_Replacement {
+	/**
+	 * Byte offset into document where replacement span begins.
+	 *
+	 * @since 6.1.0
+	 * @var int
+	 */
+	public $start;
+
+	/**
+	 * Byte offset into document where replacement span ends.
+	 *
+	 * @since 6.1.0
+	 * @var int
+	 */
+	public $end;
+
+	/**
+	 * Span of text to insert in document to replace existing content from start to end.
+	 *
+	 * @since 6.1.0
+	 * @var string
+	 */
+	public $text;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param integer $start Byte offset into document where replacement span begins.
+	 * @param integer $end   Byte offset into document where replacement span ends.
+	 * @param string  $text  Span of text to insert in document to replace existing content from start to end.
```

Comment:
```suggestion
	 * @param int    $start Byte offset into document where replacement span begins.
	 * @param int    $end   Byte offset into document where replacement span ends.
	 * @param string $text  Span of text to insert in document to replace existing content from start to end.
```
Also, the `$text` (param + prop) description might be more concisely described as "The replacement text.", if appropriate.

Reply:
let's see what @adamziel thinks about the `$text` prop. I like the extra context which makes clear the role of that parameter. when using an IDE that understands these docblock comments having the usage information inline is helpful at least to me.

plus I kind of get annoyed when I have repetition without more useful context:

```php
new WP_HTML_Text_Replacement( $start, $end, $text );
                                            ^^^^^
                                              ? what is this
                                              A: "replacement text"
```

but I agree it's a bit wordy and I think we can improve this (and all) docblocks before merge. I'm hoping to review all comments this week.

Reply:
I love how concise "the replacement text," but @dmsnell has a point that it only makes sense in context of the other two params. How about "Text to replace the existing contents between $start and $end"?

Reply:
I've been thinking of this all day. it could be the case that this entire class could use some major naming updates that could remove this awkwardness. classes like these have an extraordinary impact on performance (unfortunately), but they are awkward.